File: bestia-messages/src/main/java/net/bestia/messages/inventory/InventoryItemUseMessage.java
package net.bestia.messages.inventory;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import net.bestia.messages.AccountMessage;
import net.bestia.messages.InputMessage;

/**
 * Advises the server to use an item from the inventory of the player
 * 
 * @author Thomas Felix <thomas.felix@tfelix.de>
 *
 */
public class InventoryItemUseMessage extends InputMessage {

	@JsonIgnore
	private static final long serialVersionUID = 1L;

	public final static String MESSAGE_ID = "inventory.item.use";

	@JsonProperty("iid")
	private int itemId;

	/**
	 * Std. Ctor.
	 */
	public InventoryItemUseMessage() {

	}

	public InventoryItemUseMessage(long accId, int pbid) {
		this.setAccountId(accId);
		this.setPlayerBestiaId(pbid);
	}

	public InventoryItemUseMessage(AccountMessage msg, int pbid) {
		super(msg, pbid);
	}

	public int getItemId() {
		return itemId;
	}

	@Override
	public String getMessageId() {
		return MESSAGE_ID;
	}

	@Override
	public String getMessagePath() {
		return getZoneMessagePath();
	}

	@Override
	public String toString() {
		return String.format("InventoryItemUseMessage[accId: %d, bestiaId: %d, itemId: %d]", getAccountId(),
				getPlayerBestiaId(), itemId);
	}

}

File: bestia-messages/src/main/kotlin/net/bestia/messages/inventory/InventoryItemUseMessage.kt
package net.bestia.messages.inventory

import net.bestia.messages.EntityMessage

/**
 * Signals the server to use an castable item on the map possibly spawning map
 * entities or doing things to the map/zone itself.
 *
 * @author Thomas Felix
 */
data class InventoryItemUseMessage(
    override val entityId: Long,

    val playerItemId: Int,
    /**
     * Token for identifying the cast request on the client and receive the
     * confirm message.
     */
    val token: String,
    val x: Long = 0,
    val y: Long = 0,
    val z: Long = 0
) : EntityMessage
